### PR TITLE
alac: Add the Apple Lossless Audio Codec (ALAC) library 'libalac'

### PR DESCRIPTION
**Pull request by mikebrady**
This patch adds a the 'alac' package to OpenWrt. The package builds 'libalac',
a library that contains the Apple Lossless Audio Codec (ALAC).

The code is from a fork of the source published by Apple at
http://alac.macosforge.org.

The fork used is at https://github.com/mikebrady/alac.

Libalac can optionally be used by Shairport Sync.

Signed-off-by: Mike Brady <mikebrady@eircom.net>

### CODE REVIEW

**Comment by thess** [comment]
Please remove the 2 license files from this submission. If they are necessary, they should be in the source package.

Will there be a future Shairport Sync which actually uses this package? Current version has built-in decoder (I think).

**Comment by mikebrady** [comment]
Hi Ted. Thanks for the feedback. So I went back to the original Apple stuff looking for license information, and since I was last there they have put up their own ALAC repository which can produce a static library without modification. So I'm going to withdraw this request and look at their new-to-me stuff.

As things stand, Shairport Sync 3.0 can use the ALAC decoder instead of the built-in decoder made by David Hammerton.

**Comment by mikebrady** [comment]
Hello again Ted. The Apple ALAC package has two licences in it – an Apple Public Source License V2 licence and an Apache 2 license version 2. Should I just note that in the  PKG_LICENSE line?

Also, what would be preferable for OpenWrt, a dynamic library or a static one? The libalac library is quite small in any case, around 30k, but I wonder what you think.

**Comment by thess** [comment]
Hi Mike -
The PKG_LICENSE and PKG_LICENSE_FILES definitions are indeed a list - look at ffmpeg for example.

The dynamic lib should be in the distribution package. However, it's OK to provide both in the _installdev_ step in case some other application wants to link it statically.

**Comment by mikebrady** [comment]
Hi there. I'd like to reopen the request please – I prefer my repository to the Apple one, which looks dead. I've fixed the issues you referred to and modified the *installdev* package, including the static library and the `.la` file.

**Comment by antonlacon** [comment]
Hello,

The PKG_LICENSE_FILES line within the Makefile should be there. This lists the licensing files for someone to know what they're called.

Please rebase your branch against master and squash the patches down to a single patch submission.

File: libs/alac/Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (C) 2016 OpenWrt.org
```
Comment (diizzyy) [inline libs/alac/Makefile]: Don't assign copyright to OpenWrt unless you're working under contract or affiliated.

**Comment by diizzyy** [comment]
Wouldn't it be better to use upstream repo and have autoconf-support added as a patch?

File: libs/alac/Makefile (hunk 1 above)
Comment (mikebrady) [inline libs/alac/Makefile]: Thanks. I thought it was standard practice to do this.

**Comment by mikebrady** [comment]
Well, if it was just being packaged up for OpenWrt, your suggestion would be perfect, but it's for other distributions as well. And I've been as respectful as I could of the original repository.  BTW I've put in a request to update what appears to be the "official" repository.

File: libs/alac/Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=alac
+PKG_VERSION:=0.0.7
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/mikebrady/alac
```
Comment (hauke) [inline libs/alac/Makefile]: Please use the official release or repository as a source and add patches if needed.

File: libs/alac/Makefile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libalac.la $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libalac.a $(1)/usr/lib/
+	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/ $(1)/usr/lib/pkgconfig/
+endef
+
+define Package/libalac/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libalac.so* $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libalac.la $(1)/usr/lib/
```
Comment (hauke) [inline libs/alac/Makefile]: Please do not package *-la files here.

**Comment by diizzyy** [comment]
http://perkele.cc/software/ALAC
This contains bug fixes and performance enhancements we want added if we're going to merge this most likely.

**Comment by mikebrady** [comment]
Many thanks – I wasn't aware of this work.

**Comment by champtar** [comment]
as the original repo as been deleted, I can't reopen the PR (sorry)

**Comment by mikebrady** [comment]
Apologies – I've been neglecting this, developing Shairport Sync itself instead. I'll return to this in a while. Thanks for all the suggestions.

**Pull request closed without merging**